lib/universal-search.js

```diff
@@ -32,6 +34,7 @@ Search.prototype = {
     win.universalSearch = win.universalSearch || {}
 
     this._hideSearchBar(win);
+    this._enableSearchSuggestions(win);
```

Comment:
Hmm. The code inside `loadIntoWindow` runs in every XUL window, but because prefs are global, and we only need to adjust this pref on install / uninstall, I think we only want to run this code once, as part of the install/uninstall process. I'm not 100% sure about this, but it looks like the about:config page [flushes changes to disk](https://dxr.mozilla.org/mozilla-central/source/mobile/android/chrome/content/config.js#160-161) using `Services.prefs.savePrefFile(null)`. I bet that approach will work here; it seems to be used in some other important-seeming spots when I search for it in dxr.

In order to run on install but not regular startup, the bootstrap.js `startup` method needs to be updated to check that `reason` is ADDON_INSTALL or ADDON_ENABLE (the constants are explained [here](https://developer.mozilla.org/en-US/Add-ons/Bootstrapped_extensions#Reason_constants)), and if it's one of those reasons, call `UniversalSearch.enableSearchSuggestions`.

I think we'll also need to add similar checks in the bootstrap.js `unload` method. Unfortunately, I'm not sure where we can put the user's original preference such that it persists across restarts (I don't think setting it on the window will work, though it seemed to work in the old addon; I could be wrong here). We could either try to figure out how to save this in web storage, or we could just unset the preference entirely, via [`Services.prefs.clearUserPref`](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIPrefBranch#clearUserPref()).